packages/block-library/src/navigation/edit/index.js

```diff
@@ -279,14 +279,24 @@ function Navigation( {
 		setIsPlaceholderShown( ! isEntityAvailable );
 	}, [ isEntityAvailable ] );
 
+	// If the ref no longer exists the reset the inner blocks
+	// to provide a clean slate.
+	useEffect( () => {
+		if ( ref === undefined && innerBlocks.length > 0 ) {
+			replaceInnerBlocks( clientId, [] );
+		}
+		// innerBlocks are intentionally not listed as deps. This function is only concerned
+		// with the snapshot from the time when ref became undefined.
+	}, [ clientId, ref, innerBlocks ] );
```

Comment:
I think this has caused a potential bug around pattern insertion in the nav block, the nav block no longer supports UnsavedInnerBlocks. Every time the block has Inner Blocks but no ref, the inner blocks are deleted straight away.

It is a little hard to follow, so I fully appreciate the difficulty level in understanding why this would be a problem.

Also just wanted to flag the comment - 'innerBlocks are intentionally not listed as deps.' - they are listed as deps?

Reply:
I've created a bug report for this - https://github.com/WordPress/gutenberg/issues/37354

Reply:
Thanks Dan and apologies for any problems.

I guess this is why we need stable tests.